### PR TITLE
2357 add junctions to suburban widgets - step 1

### PR DESCRIPTION
**Pull request by ziv17**
Add road-junction-km table

### CODE REVIEW

File: alembic/versions/7ea883c8a245_add_road_junction_km_table.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                    )
+    op.create_index('road_junction_km_idx',
+                    "road_junction_km",
+                    ['road', 'non_urban_intersection'],
+                    unique=True)
+
+
+# noinspection PyUnresolvedReferences
+def downgrade():
+    op.drop_index(op.f('road_junction_km_idx'), table_name="road_junction_km")
```
Comment (BarVolunteering) [inline alembic/versions/7ea883c8a245_add_road_junction_km_table.py:35]: If you use 'op.f' here do you need to use it also in create_index?

File: anyway/models.py [hunk 2]
```diff
@@ -1242,6 +1242,21 @@ def serialize(self):
         }
 
 
+class RoadJunctionKM(Base):
+    __tablename__ = "road_junction_km"
+    MAX_NAME_LEN = 100
+    road = Column(Integer(), primary_key=True, nullable=False)
```
Comment (BarVolunteering) [inline anyway/models.py:1248]: BigInteger?

**Comment by atalyaalon** [comment]
@BarVolunteering thanks for reviewing. @ziv17 what do you think?

File: anyway/infographics_utils.py [hunk 3]
```diff
@@ -262,3 +263,47 @@ def get_infographics_mock_data():
...
+            rkj[t.road][t.km] = t.non_urban_intersection
+        tmp: List[RoadSegments] = db.session.query(RoadSegments).all()
+        segments = {t.segment_id: t for t in tmp}
+        for k, v in segments.items():
+            if v.road not in rkj:
+                logger.warning(f"No junctions in road {v.road}.")
+                continue
+            # logger.debug(f"seg:{k},road:{v.road}:juncs:{rkj[v.road]}.")
+            junctions = [rkj[v.road][km] for
+                         km in rkj[v.road].keys() if v.from_km-1 <= km <= v.to_km+1]
```
[the remark this answers is not in the document]
Comment (ziv17) [reply anyway/infographics_utils.py:306]: Checking... (אני צריך להיזכר)

[the remark this answers is not in the document]
Comment (atalyaalon) [reply anyway/infographics_utils.py:306]: Thanks @ziv17 !
I've taken a look at the first 3, seems like they are within the segment
see [this path](https://goo.gl/maps/g1muMZcVEE4hgN7H8) and see [the segment in our infographics](https://media.anyway.co.il/location/10070)

Regarding non urban intersection 10058 - see [this redash query](https://redash.dataforchange.org.il/queries/26/source) I downloaded the data and took the longitude, latitude which are:  latitude = 31.797481265456900, longitude=35.14807657087000, if you put it [on google maps](https://goo.gl/maps/dWEk9KgJHRmT38S18) it really close to Harel Junction.

Regarding non urban intersection 31 -  "צומת עין חמד" - see [this redash query](https://redash.dataforchange.org.il/queries/27/source) I downloaded the csv from redash and took the longitude, latitude which are:  latitude = 31.800648410974300, longitude=35.133438643819300, if you put it [on google maps](https://goo.gl/maps/Qhht3gsgqFEENrMo7) it's indeed within the segment כביש 1 במקטע מחלף שורש - מחלף הראל.

All in all seems about right, what did I miss?

File: anyway/models.py (hunk 2 above)
Comment (atalyaalon) [reply anyway/models.py:1248]: I agree, Integer is enough here, unless I missed anything :)

**Comment by atalyaalon** [comment]
Also, @ziv17 small note to remember before we'll merge the pr, alembic schema has changed, hence it's better to update down revision of the alembic version file before merging

File: anyway/infographics_utils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (atalyaalon) [reply anyway/infographics_utils.py:306]: @ziv17 re-read this and I understand that by my method Harel Junction won't be inside.
I'll ask from the CBS the updated table of junctions, and will ask them about this use case, I will let you know ASAP

**Pull request closed without merging**